app/src/main/java/org/openedx/app/AppActivity.kt

```diff
@@ -49,6 +50,15 @@ class AppActivity : AppCompatActivity(), InsetHolder, WindowSizeHolder, AppDataH
 
     private var _windowSize = WindowSize(WindowType.Compact, WindowType.Compact)
 
+    private val authCode: String?
+        get() {
+            val data = intent?.data
+            if (data is Uri && data.scheme == BuildConfig.APPLICATION_ID && data.host == "oauth2Callback") {
+                return data.getQueryParameter("code")
+            }
+            return null
+        }
+
```

Comment:
If the application was opened using the auth intent filter, 
`<applicationid>://oauth2Callback?code=<some_code>` it will get the code out of the URL and return that.